vault/expiration.go

```diff
@@ -325,6 +328,14 @@ func getNumExpirationWorkers(c *Core, l log.Logger) int {
 	return numWorkers
 }
 
+// alsoSendEvent returns an ExpireLeaseStrategy that sends an expiration event after executing the given ExpireLeaseStrategy.
+func alsoSendEvent(e ExpireLeaseStrategy) ExpireLeaseStrategy {
+	return func(ctx context.Context, m *ExpirationManager, leaseID string, namespace *namespace.Namespace) {
+		e(ctx, m, leaseID, namespace)
```

Comment:
Are there any failure conditions for `e` that we should include/emit in the expiration event? Or don't send the event if e fails?
If not, are there any trade-offs between starting the go routine before or after `e`?

Reply:
I think the right thing to do is to *not* send the event if `e` panics, but there's not currently a way to know if it failed other than that.

The benefit to starting after is that if `e` does panic, then the event won't be sent.

The biggest disadvantage of this approach is that we have to fetch the lease from storage again when we send the event, which costs a little bit of performance. We could fix this by moving the event sending into the `e` function itself, but this would require modifying each of the `ExpireLeaseStrategy` implementations, which is more work to maintain. Or we could rework the `ExpireLeaseStrategy` interface so we could avoid the additional fetch of the lease.

Reply:
We may want to draw a distinction between expiration and revocation.  In other words, what we're doing now in this PR is firing an event saying a lease expired, but ignoring whether or not the lease expiration succeeded, and whether or not the associated dynamic secrets, if any, got revoked.